### ISSUE TITLE
Bulk crashed

### ISSUE THREAD

**Issue by forchain** [dgraph-io/dgraph#1786]
goroutine 55733 [runnable]:
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/table.(*Table).readIndex.func1(0xcd35a52000, 0xcaebd821c0, 0xc612204b40)
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/table/table.go:265 +0x191
created by github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/table.(*Table).readIndex
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/table/table.go:251 +0x554

goroutine 49732 [semacquire, 1 minutes]:
sync.runtime_Semacquire(0xccc66292ac)
        /home/travis/.gimme/versions/go1.9.2.linux.amd64/src/runtime/sema.go:56 +0x39
sync.(*WaitGroup).Wait(0xccc66292a0)
        /home/travis/.gimme/versions/go1.9.2.linux.amd64/src/sync/waitgroup.go:131 +0x72
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*DB).batchSetAsync.func1(0xccc6629270, 0xcf001192d0)
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/db.go:676 +0x35
created by github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*DB).batchSetAsync
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/db.go:675 +0x9c

Some information:
Version: 0.9
Memory: 64G (32x2)
CPU: 2 CPU (16 core each)
SSD: A :480G, B:1T. RDF stored in A, tmp and out folder stored in B
Data size: around 3 billion RDFs , 10 billion edges

**Comment by peterstace** [comment]
Hi @forchain , thank you for reporting. Do you have the rest of the stack trace? The trace for the goroutine that caused the crash will be the most useful, although the whole thing would be ideal.

**Comment by forchain** [comment]
Hi Peter, the whole call stack is too long and exceeds the terminal buffer so I could not paste them all. However, I found the are the same with the two I pasted (possible I missed some since I didn't check them very carefully), so I just pasted a small piece.
If I meet again, I will try to paste as more as possible.

Another information, I guess the problem  may be related to the number of go routine parameter because when I reduced it from 32 to 16, I didn't meet the problem again.
